src/dynamics/geometry.jl

```diff
@@ -7,22 +7,22 @@ Construct Geometry struct containing parameters and arrays describing an iso-lat
 and the vertical levels. Pass on `SpectralGrid` to calculate the following fields
 $(TYPEDFIELDS)
 """
-@kwdef struct Geometry{NF<:AbstractFloat, Grid<:AbstractGrid} <: AbstractGeometry
+@kwdef struct Geometry{
+    NF,
+    Grid,
+    VectorType,
```

Comment:
I am not even sure we need everything from the Geometry on GPU for the GPU model. 

In principle, indices stay on the CPU, even when it's a GPU model. And indices are sometimes derived from the geometry, aren't they? 

Of course, it doesn't hurt to write the struct like that, and in the end we can still just always initialise with `Array` if that's what we want to use.

Reply:
Yeah, for `Geometry` I don't think it's necessary to move things to the GPU, but actually the `coslat` vectors would be good. For the interpolation though, we'll need that as it's actually "data".